api/api/v1alpha1/manifest_types.go

```diff
@@ -38,7 +48,8 @@ type ChartInfo struct {
 
 // ManifestSpec defines the specification of Manifest
 type ManifestSpec struct {
-	Charts []ChartInfo `json:"charts,omitempty"`
+	Charts          []ChartInfo      `json:"charts,omitempty"`
+	CustomResStates []CustomResState `json:"customResStates,omitempty"`
```

Comment:
why do you want to have a state pushed into the spec? Wouldn't that be a bitter fit for status?